### ISSUE TITLE
Deadlock when ticker stopped with no readers on channel

### ISSUE THREAD

**Issue by zwass** [mixer/clock#6]
The following test consistently causes a deadlock:
```go
func TestTickerDeadlock(t *testing.T) {
	c := NewMockClock()
	tk := c.NewTicker(5 * time.Millisecond)
	c.AddTime(6 * time.Millisecond)
	time.Sleep(1 * time.Millisecond)
	tk.Stop()
}
```

Per the [time godoc](https://godoc.org/time#NewTicker), the ticker may "drops tick to make up for slow receivers". We probably need to do this to avoid the deadlock here.

**Comment by connor4312** [comment]
Thanks for the issue! Fixed in #6 https://github.com/mixer/clock/commit/ac3776ffbf764bc40bda24e84184146865d0a46d

I opened not to go with the "drop ticks" approach. The reason is that I want this package to be as deterministic as possible. Dropping ticks after a certain real time introduces a degree of non-determinism. One approach might be to add a third `case` in that statement, like `case <-m.clock.After(TickerDropInterval)`, where the drop time is actually advanced by the package consumer. Thoughts?
